src/core/toga/widgets/divider.py

```diff
@@ -0,0 +1,51 @@
+from toga.style.pack import ROW
+
+from .base import Widget
+
+
+class Divider(Widget):
+    """A visual divider line.
+
+    Args:
+        id (str): An identifier for this widget.
+        style (:obj:`Style`): An optional style object. If no style is provided then
+            a new one will be created for the widget.
+        direction: The direction for divider, either `Divider.AUTO`, `Divider.HORIZONTAL`
+            or `Divider.VERTICAL`. In case of `Divider.AUTO`, the direction will be
+            determined from the layout when the divider is added to a `toga.Box`.
+        factory (:obj:`module`): A python module that is capable to return a
+            implementation of this class with the same name. (optional & normally not needed)
+    """
+    AUTO = 0
```

Comment:
I'm not sure an "AUTO" direction makes sense here. 

The "determine direction from parent's layout direction" is a neat trick, but you can't assume that the style properties will be Toga's Pack layout properties. The style language is deliberately abstracted; the style can control widgets, but not vice versa. If this technique were to be applied, it would need to come *from* the style language, and be applied *on* the widget.

Even if we could make this work - I'm not sure I can see a use for it. The use case here would be a separator that changes from horizontal to vertical depending on how it's parent widgets are laid out; I'm not sure that's a realistic use case. It would certainly be a neat trick, but I'm not sure it's worth the complexity supporting it would entail.